### ISSUE TITLE
Feature request: import/exports-first rule (opposite of exports-last)

### ISSUE THREAD

**Issue by schickling** [oxc-project/oxc#17706]
## Summary

I'd like to request a new rule `import/exports-first` that enforces exported members to be declared **before** non-exported members in a file. This is the opposite of the existing `exports-last` rule.

## Motivation

In some codebases, the convention is to place the public API (exported members) at the top of a file, making it immediately visible what the module exposes. Non-exported implementation details follow below. This improves readability by putting the "interface" first.

**Pattern enforced by this rule:**

```ts
// ✅ Correct: exports first
export const publicApi = () => { /* ... */ }
export const anotherExport = 42

const helperFunction = () => { /* ... */ }
const internalState = {}
```

```ts
// ❌ Incorrect: non-exported before exported
const helperFunction = () => { /* ... */ }

export const publicApi = () => { /* ... */ }
```

## Comparison with exports-last

| Rule | Pattern |
|------|---------|
| `exports-last` | Implementation first, exports at bottom |
| `exports-first` (proposed) | Exports at top, implementation after |

Both are valid conventions depending on team preference. Having both rules available lets teams choose their preferred style.

## Additional context

- This would be a stylistic rule similar to `exports-last`
- Could potentially share implementation logic with `exports-last` (inverted check)

---

*This issue was created on behalf of a user by an AI assistant.*

**Comment by camc314** [comment]
Thanks for the suggestion! This is an interesting stylistic rule.

Since this is a stylistic rule, and not part of the core feature set, I think this would fit in better as a custom JavaScript plugin rather than being built into the core linter. Here's why:
1. Stylistic preference: Both exports-first and exports-last are purely stylistic conventions, and teams often have different preferences
2. Low complexity: The rule logic is straightforward and can be easily implemented as a custom plugin
3. Keeps the core lean: Adding every stylistic variation to the core increases maintenance burden

Here's how you could implement it :

FIrst, create your plugin:
```ts
// exports-first-plugin.ts
import { definePlugin } from "oxlint";
export default definePlugin({
  meta: { name: "custom" },
  rules: {
    "exports-first": {
      meta: {
        type: "suggestion",
        docs: {
          description: "Enforce exports are declared before non-exported members",
        },
      },
      create(context) {
        let lastExportIndex = -1;
        let firstNonExportIndex = -1;
        const violatingExports = [];
        return {
          Program(node) {
            const body = node.body;
            // Find the first non-export and collect all exports that come after it
            for (let i = 0; i < body.length; i++) {
              const statement = body[i];
              const isExport =
                statement.type === "ExportNamedDeclaration" ||
                statement.type === "ExportDefaultDeclaration" ||
                statement.type === "ExportAllDeclaration";
              // Skip import declarations - they're typically allowed anywhere at the top
              if (statement.type === "ImportDeclaration") continue;
              if (isExport) {
                if (firstNonExportIndex !== -1) {
                  // Found an export after a non-export statement
                  violatingExports.push(statement);
                }
                lastExportIndex = i;
              } else {
                if (firstNonExportIndex === -1) {
                  firstNonExportIndex = i;
                }
              }
            }
            // Report all exports that appear after non-exports
            for (const stmt of violatingExports) {
              context.report({
                node: stmt,
                message:
                  "Export statements should appear before non-exported members",
              });
            }
          },
        };
      },
    },
  },
});
```

Second add the plugin to your `.oxlintrc.json`

```ts
{
  plugins: {
    custom: ./exports-first-plugin.js
  },
  rules: {
    custom/exports-first: error
  }
}
```
